lib/src/utils.rs

```diff
@@ -147,26 +147,37 @@ pub fn generate_transactions_for_batch_blocks(
         ),
         "only pacaya batch supported"
     );
+    assert!(
+        taiko_guest_batch_input.tx_data_from_calldata.is_empty()
```

Comment:
```suggestion
        taiko_guest_batch_input.tx_data_from_calldata.is_empty() != taiko_guest_batch_input.tx_data_from_blob.is_empty()
```
use `!=` check `not both`